docs/examples/nlp_sentiment.md

```diff
@@ -0,0 +1,1399 @@
+                "result": [888, 1, 1156, 0, 0, 0, 0, 0, 0, 0],
+            },
+        ]
+    }
+
+    def _load(self):
+        self.vocabulary = {}
+        with open(self.asset_path, "r", encoding="utf-8") as f:
+            for i, k in enumerate(f):
+                self.vocabulary[k.strip()] = i + 2
+        self._vectorizer = np.vectorize(lambda x: self.vocabulary.get(x, 1))
+
+    def _predict(self, tokens, length=None, drop_oov=True):
+        vectorized = (
+            np.array(self._vectorizer(tokens), dtype=np.int)
+            if tokens
+            else np.array([], dtype=int)
+        )
+        if drop_oov and len(vectorized):
+            vectorized = np.delete(vectorized, vectorized == 1)
+        if not length:
+            return vectorized.tolist()
+        result = np.zeros(length)
+        vectorized = vectorized[:length]
+        result[: len(vectorized)] = vectorized
+        return result.tolist()
+
+
+class MovieReviewItem(pydantic.BaseModel):
+    text: str
+    rating: Optional[
+        float
+    ] = None  # could be useful in the future ? but not mandatory
+
+
+class MovieSentimentItem(pydantic.BaseModel):
+    label: str
+    score: float
+
+
+class Classifier(modelkit.Model[MovieReviewItem, MovieSentimentItem]):
+    CONFIGURATIONS = {
+        "imdb_classifier": {
+            "asset": "imdb_model.h5",
+            "model_dependencies": {
+                "tokenizer": "imdb_tokenizer",
+                "vectorizer": "imdb_vectorizer",
+            },
+        },
+    }
+    TEST_CASES = {
+        "cases": [
+            {
+                "item": {"text": "i love this film, it's the best I've ever seen"},
+                "result": {"score": 0.8441019058227539, "label": "good"},
+            },
+            {
+                "item": {"text": "this movie sucks, it's the worst I have ever seen"},
+                "result": {"score": 0.1625385582447052, "label": "bad"},
+            },
+        ]
+    }
+
+    def _load(self):
+        self.model = tf.keras.models.load_model(self.asset_path)
+        self.tokenizer = self.model_dependencies["tokenizer"]
+        self.vectorizer = self.model_dependencies["vectorizer"]
+        self.prediction_mapper = np.vectorize(lambda x: "good" if x >= 0.5 else "bad")
+
+    def _predict_batch(self, reviews):
+        texts = (review.text for review in reviews)
+        cleaned_reviews = self.tokenizer.predict_batch(texts)
+        vectorized_reviews = self.vectorizer.predict_batch(cleaned_reviews, length=64)
+        predictions_scores = self.model.predict(vectorized_reviews)
+        predictions_classes = self.prediction_mapper(predictions_scores).reshape(-1)
+        predictions = [
+            {"score": score, "label": label}
+            for score, label in zip(predictions_scores, predictions_classes)
+        ]
+        return predictions
+
+
+model_library = modelkit.ModelLibrary(models=[Tokenizer, Vectorizer, Classifier])
```

Comment:
model_library = modelkit.ModelLibrary(models=[Classifier]) ?

Reply:
Since the Tokenizer and Vectorizer are dependencies of Classifier, you need to make them discoverable and add them to the `models` parameter

Reply:
Oh, so if a model a had dependencies a_1 & a_2 and model a_1 has dependencies a_1_1 & a_1_2 do I need to do 
- modelkit.ModelLibrary(models=[a]) 
- modelkit.ModelLibrary(models=[a, a_1, a_2])
- modelkit.ModelLibrary(models=[a, a_1, a_2, a_3, a_1_1, a_1_2])

Reply:
Go for the third option !

Reply:
That's why it is important and relevant to organize your Models as a package, so you do not need to list allllll your dependencies like this, but rather:
```python
from my_project import my_models
modelkit.ModelLibrary(models=my_models)
```